types/V1.mli

```diff
@@ -299,38 +302,53 @@ module type BLOCK = sig
 
 end
 
-(** {1 Network stack}
+(** {1 Network stack} *)
+
+module Network : sig
+  type error = [
+    | `Unknown of string (** an undiagnosed error *)
+    | `Unimplemented     (** operation not yet implemented in the code *)
+    | `Disconnected      (** the device has been previously disconnected *)
+  ]
+  (** The type for IO operation errors *)
+
+  (* XXX: here we'd like to have a printer!  but maybe more convenient in a V1_PP.mli+ml? *)
+
+  type stats = {
+    mutable rx_bytes: int64;
+    mutable rx_pkts: int32;
+    mutable tx_bytes: int64;
+    mutable tx_pkts: int32;
+  }
+  (** The type for frame statistics to track the usage of the
+      device. *)
+end
 
-    A network interface that serves Ethernet frames. *)
+(** A network interface that serves Ethernet frames. *)
 module type NETWORK = sig
+  open Network
 
   type page_aligned_buffer
   (** The type for page-aligned memory buffers. *)
 
   type buffer
   (** The type for memory buffers. *)
 
-  type error = [
-    | `Unknown of string (** an undiagnosed error *)
-    | `Unimplemented     (** operation not yet implemented in the code *)
-    | `Disconnected      (** the device has been previously disconnected *)
-  ]
-  (** The type for IO operation errors *)
-
   type macaddr
   (** The type for unique MAC identifiers for the device. *)
 
   include DEVICE with
     type error := error
 
-  val write: t -> buffer -> unit io
+  val write: t -> buffer -> (unit, error) result io
   (** [write nf buf] outputs [buf] to netfront [nf]. *)
 
-  val writev: t -> buffer list -> unit io
+  val writev: t -> buffer list -> (unit, error) result io
   (** [writev nf bufs] output a list of buffers to netfront [nf] as a
       single packet. *)
 
-  val listen: t -> (buffer -> unit io) -> unit io
+  (* XXX: listen currently runs forever (documentation wrong)! *)
+  val listen: t -> (buffer -> unit io) -> (unit, error) result io
```

Comment:
This `buffer -> unit io` means the function we call can't pass us any errors -- we can only emit those we caused ourselves.  When attempting to revise users to use result types, this presents some difficulties.  Is it intentional, or an opportunity for more discussion?